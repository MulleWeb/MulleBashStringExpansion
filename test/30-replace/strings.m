#import <MulleBashStringExpansion/MulleBashStringExpansion.h>

#include <stdio.h>


static NSString  *expressions[] =
{
   @"foo",
   @"foo:-baz",
   @"baz:-foo",
   @"baz:=foo",
   @"foo:3",
   @"foo:1:1",
   @"foo#bar",
   @"foo%bar",
   @"foo^^",
   @"foo^^[a-b]",
   @"foo//b/c",
   nil,
};


int   main( void)
{
   NSString       *value;
   NSDictionary   *info;
   S              *s;
   NSString       **p;

   info  = @{ @"foo": @"foobar" };
   s     = [S objectWithMulleScionLocalVariables:nil
                                      dataSource:info];
   value = [s:@"foobar"
             :@"(o+)"
             :@"x\\1x"];
   mulle_printf( "\"%@\"\n", value);

   for( p = expressions; *p; p++)
   {
      value = [s $:*p];
      mulle_printf( "\"%@\" -> \"%@\"\n", *p, value);
   }
   return 0;
}
