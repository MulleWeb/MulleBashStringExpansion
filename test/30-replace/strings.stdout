"fxooxbar"
"foo" -> "foobar"
"foo:-baz" -> "foobar"
"baz:-foo" -> "foo"
"baz:=foo" -> "foobar"
"foo:3" -> "bar"
"foo:1:1" -> "o"
"foo#bar" -> "foobar"
"foo%bar" -> "foo"
"foo^^" -> "FOOBAR"
"foo^^[a-b]" -> "fooBAr"
"foo//b/c" -> "foocar"
